File: apps/frontend/src/app/models/register-user.dto.ts
import { Dto } from './dto.base';
import { Model } from './model.type';

/**
 * Dto for registering a user.
 *
 * @property {string} username - The username of the user
 * @property {string} firstName - The first name of the user
 * @property {string} lastName - The last name of the user
 * @property {string} email - The email address of the user
 * @property {string} password - The password of the user
 */
export class RegisterUserDto extends Dto {
  public username!: string;
  public firstName!: string;
  public lastName!: string;
  public email!: string;
  public password!: string;
}

export type RegisterUser = Model<RegisterUserDto>;

File: apps/backend/src/main/java/tabletop/gather/backend/auth/RegisterUserDto.java
package tabletop.gather.backend.auth;

import jakarta.validation.constraints.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class RegisterUserDto {

  @NotEmpty(message = "Username is required")
  @Size(max = 50, message = "Username is too long")
  private String username;

  @NotEmpty(message = "First name is required")
  @Size(max = 255, message = "First name is too long")
  private String firstName;

  @NotEmpty(message = "Last name is required")
  @Size(max = 255, message = "Last name is too long")
  private String lastName;

  @NotEmpty(message = "Email is required")
  @Email(message = "Provide a valid email address")
  @Size(max = 320, message = "Email address is too long")
  private String email;

  @NotEmpty(message = "Password is required")
  @Size(min = 3, max = 64, message = "Password must be between 3 and 64 characters")
  private String password;
}
